File: app/src/main/java/com/google/firebase/ml/md/java/settings/SettingsActivity.java
/**
 * Created by Anand kashyap on 19,July,2020
 */

package com.google.firebase.ml.md.java.settings;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import com.google.firebase.ml.md.R;

/** Hosts the preference fragment to configure settings. */
public class SettingsActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    setContentView(R.layout.activity_settings);
    if (getSupportActionBar() != null) {
      getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }
    getSupportFragmentManager()
        .beginTransaction()
        .replace(R.id.settings_container, new SettingsFragment())
        .commit();
  }

  @Override
  public boolean onSupportNavigateUp() {
    onBackPressed();
    return true;
  }
}

File: app/src/main/java/com/google/firebase/ml/md/kotlin/settings/SettingsActivity.kt
/**
 * Created by Anand kashyap on 19,July,2020
 */

package com.google.firebase.ml.md.kotlin.settings

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.ml.md.R

/** Hosts the preference fragment to configure settings.  */
class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_settings)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.settings_container, SettingsFragment())
                .commit()
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}
